backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java

```diff
@@ -0,0 +1,244 @@
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+class PostControllerTest {
+
+    public MockMvc mockMvc;
+
+    @Autowired
+    PostController postController;
+
+    @MockBean
+    public PostService postService;
+
+    @BeforeEach
+    public void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(postController).build();
+    }
+
+    @Test
+    @DisplayName("Returns a list of posts at - GET /api/threads")
+    void getPosts() throws Exception {
+        // given
+        User user = new User(1L,
+                "user1",
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+        Post p1 = new Post(1L, "text", user, t1);
+
+        when (postService.getAllPosts())
+                .thenReturn(Arrays.asList(p1));
+
+        // when then
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/posts"))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().is(200))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.size()", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].text", Matchers.is("text")));
+    }
+
+    @Test
+    @DisplayName("Returns a post by ID at - GET /api/thread/{id}")
+    void getPostById() throws Exception {
+        // given
+        User user = new User(1L,
+                "user1",
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+        Post p1 = new Post(1L, "text", user, t1);
+
+        when (postService.getPostById(any()))
+                .thenReturn(p1);
+
+        // when then
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/post/1"))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().is(200))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(1)))
```

Comment:
Maybe this is where we should check that length is definitely 1? And we're not accidentally getting back multiple items? Might be overkill, of course 😄

Reply:
Fair enough